### PR TITLE
Add range limit option for `range` selection mode

### PR DESCRIPTION
**Pull request by IsabellaPap**
:sparkles: :nail_care: 

## Changes

- add `rangeLimit` option to `date-time-inline`
- set `maxDate` dynamically to +`rangelimit` days when the first date is selected in `range` mode.

### CODE REVIEW

File: projects/picker/src/lib/date-time/date-time-inline.component.ts [hunk 1]
```diff
@@ -199,6 +199,17 @@ export class OwlDateTimeInlineComponent<T> extends OwlDateTime<T> implements OnI
     }
   }
 
+  /**
+   * Limit to the amount of days that can be selected at once.
+   * */
```
Comment (NetWin) [inline projects/picker/src/lib/date-time/date-time-inline.component.ts:204]:
```suggestion
   */
```

**Review by NetWin — CHANGES_REQUESTED** [review]
Awesome! Thank you. Would you mind adding the range limit to the ng-doc so we can display it's use/functionality there aswell? 😊

File: projects/picker/src/lib/date-time/date-time-inline.component.ts (hunk 1 above)
Comment (IsabellaPap) [reply projects/picker/src/lib/date-time/date-time-inline.component.ts:204]:
```suggestion
  /**
   * Limit to the amount of days that can be selected at once.
   * */
  @Input()
  public rangeLimit: number | null = null;

  /**
   * Variable to hold the old max date time value for when we override it with rangeLimit
   * */
  private _oldMaxDateTime: T | null;

  /**
   * Emits selected year in multi-year view
   * This doesn't imply a change on the selected date.
   * */
  @Output()
  yearSelected = new EventEmitter<T>();
```
All the other ones have 2 so maybe I should just add to the one on line 210?

Comment (NetWin) [reply projects/picker/src/lib/date-time/date-time-inline.component.ts:204]: Yuck 🙈 
I'll note that down and fix it soon. Leave it as is then.

**Review by NetWin — APPROVED** [review]
Ignore my comment, ng-doc picked it up automagically 🪄 
Just played around with it a bit, Looks good to me 👍

File: projects/picker/src/lib/date-time/date-time-inline.component.ts [hunk 2]
```diff
@@ -199,6 +199,17 @@ export class OwlDateTimeInlineComponent<T> extends OwlDateTime<T> implements OnI
...
+  @Input()
+  public rangeLimit: number | null = null;
+
+  /**
+   * Variable to hold the old max date time value for when we override it with rangeLimit
+   * */
+  private _oldMaxDateTime: T | null;
```
Comment (pawcoding) [inline projects/picker/src/lib/date-time/date-time-inline.component.ts]: I would maybe rename this to `_initialMaxDateTime`

File: projects/picker/src/lib/date-time/date-time-inline.component.ts [hunk 3]
```diff
@@ -311,6 +322,19 @@ export class OwlDateTimeInlineComponent<T> extends OwlDateTime<T> implements OnI
     } else {
       this.value = date;
     }
+
```
Comment (pawcoding) [inline projects/picker/src/lib/date-time/date-time-inline.component.ts:325]: Also maybe add a comment here about what your new logic does. The whole component is already complex enough 😅

**Review by NetWin — APPROVED** [review]
🚀

**Pull request merged**